### PR TITLE
Java Lib typed records

### PR DESCRIPTION
**Pull request by saldoukhov**

### CODE REVIEW

File: sdk/javascript/packages/core/src/keeper.ts [hunk 1]
```diff
@@ -228,11 +228,17 @@ export const getClientId = async (clientKey: string): Promise<string> => {
     return platform.bytesToBase64(clientKeyHash)
 }
 
-export const initializeStorage = async (storage: KeyValueStorage, clientKey: string, domain: string | 'keepersecurity.com' | 'keepersecurity.eu' | 'keepersecurity.au') => {
+export const initializeStorage = async (storage: KeyValueStorage, clientKey?: string, domain?: string | 'keepersecurity.com' | 'keepersecurity.eu' | 'keepersecurity.au') => {
```
Comment (maksimu) [inline sdk/javascript/packages/core/src/keeper.ts:231]: also need to add GOV Cloud `govcloud.keepersecurity.us` 

In python we have [THIS COOL](https://github.com/Keeper-Security/secrets-manager/blob/master/sdk/python/core/keepercommandersm/helpers.py#L20) function that gives you ability to get servers by abbreviation (ex. `AU`) . Maybe we should have something like this in other SDKs?

File: examples/java/hello-secret/src/main/java/hello/App.java [hunk 2]
```diff
@@ -41,8 +30,8 @@ public static void main(String[] args) {
 
             // get the password from the first record
             KeeperRecord firstRecord = secrets.getRecords().get(0);
-            String firstRecordPassword = getRecordField(firstRecord, "password");
-            System.out.println(firstRecordPassword);
+            List<String> firstRecordPassword = ((Password) firstRecord.getData().getField(Password.class)).getValue();
```
Comment (maksimu) [inline examples/java/hello-secret/src/main/java/hello/App.java:33]: became a bit verbose.

Comment (saldoukhov) [reply examples/java/hello-secret/src/main/java/hello/App.java:33]: But strongly typed. Blame Java.

Comment (maksimu) [reply examples/java/hello-secret/src/main/java/hello/App.java:33]: Yeah, this is how we have our data in the record.

File: sdk/javascript/packages/core/src/keeper.ts (hunk 1 above)
Comment (maksimu) [reply sdk/javascript/packages/core/src/keeper.ts:231]: here is the link to the function: https://github.com/Keeper-Security/secrets-manager/blob/master/sdk/python/core/keepercommandersm/helpers.py#L20

**Pull request merged**